### ISSUE TITLE
DotNet CLI tools fail when using the full framework

### ISSUE THREAD

**Issue by GrantErickson** [dotnet/cli#3274]
I am porting a dnx command line tool from RC1 to RC2 with dotnet. It uses the full framework 4.5.1. I have stripped it down to a very simple project. My suspicion is that I am missing something basic, but I can't seem to find parallel examples. Apologies in advance if this is a simple fix.

## Steps to reproduce
Here is my project.json file.

    {
      "version": "1.0.2-alpha001",
      "description": "Tools.Cli Class Library",
      "authors": [ "Test" ],
      "buildOptions": {
        "emitEntryPoint": true,
        "outputName": "dotnet-hello"
      },
      "dependencies": {
        "Microsoft.DotNet.Cli.Utils": "1.0.0-preview1-002702"
      },
      "frameworks": {
        "net451": { }
      }
    }

Here is my program.cs file.

    using System;
    namespace Tools.Cli
    {
        public class Program
        {
            public static int Main(string[] args)
            {
                Console.WriteLine("\x1B[31m" + "Hello World" + "\x1B[39m");
                // The following line causes the error. Commenting out this line allows this to run.
                Console.WriteLine(Microsoft.DotNet.Cli.Utils.AnsiColorExtensions.Red("Test"));
                return 0;
            }
        }
    }

I am creating a NuGet package and referencing that from another project as a tool.

## Expected  behavior
run `dotnet hello`
Prints Hello World in red and then Test in red.

This works perfectly without the line to print Test. 

## Actual behavior
The following exception is shown. Hello World is not printed. 

    dotnet hello
    Unhandled Exception: System.IO.FileNotFoundException: 
        Could not load file or assembly 'Microsoft.DotNet.Cli.Utils, Version=1.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60' or one of its dependencies. 
        The system cannot find the file specified.
        at Tools.Cli.Program.Main(String[] args)

At this point I get an error dialog to prompt for debugging. 

I have tried this with a number of other assemblies with the same results.

## Environment data
.NET Command Line Tools (1.0.0-preview1-002702)

Product Information:
 Version:     1.0.0-preview1-002702
 Commit Sha:  6cde21225e

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.10586
 OS Platform: Windows
 RID:         win10-x64

**Comment by piotrpMSFT** [comment]
At this point we only support portable tools built on netcoreapp1.0. The error message should be more meaningful, however.

**Comment by GrantErickson** [comment]
Thank you for the response. Any idea on when/if this might be coming?

**Comment by piotrpMSFT** [comment]
there is not a plan for this today. The decision was quite intentional. Tools should be portable so that the decision to use a tool does not influence an app's target frameworks.

Does this block your tool in some way? Why do you require desktop?

**Comment by GrantErickson** [comment]
Thanks for the reply.

We built tooling for DNX that used the full framework. It has references to other full framework libraries. We tried an initial migration to .net Core, but there are lots of methods that would need to be rewritten to support the subset of functionality in .net Core. For example much of our reflection code would have to be changed. 

Additionally, as far as I know there isn't a good debug story for .net Core CLI tooling. So keeping with what works would be great. 

We have no intention of going cross platform at this point, and the level of investment required is significant. So we have a situation where we spend money for no tangible customer gain other than to get off RC1.

At this point, I think we will try to go with a console application (straightforward path with a debug story). I understand why, but it is disappointing how different the DNX RC1 was from the CLI RC2 implementation. Being an early adopter has proven quite costly.
